src/Admin/Product/Attributes/AttributesTab.php

```diff
@@ -181,7 +181,7 @@ protected function get_form( WC_Product $product ): AttributesForm {
 	 *
 	 * @return array of WooCommerce product types (e.g. 'simple', 'variable', etc.)
 	 */
-	protected function get_applicable_product_types(): array {
+	public static function get_applicable_product_types(): array {
```

Comment:
Given where this method is used, it might make more sense to have this provide a key-based array, rather than a value-based array. When called in `get_hidden_product_types()`, we can then use `array_diff_key()` to compare. And when this is called from `VariationsAttrubutes::register()`, we can use `array_key_exists()` as a more efficient comparison.

That will require some small tweaks to the `add_tab()` method, but I think it would be beneficial to switch to that approach.